integration_test/scenarios/search_result_highlights_scenario.dart

```diff
@@ -0,0 +1,85 @@
+import 'package:core/core.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+import '../base/base_scenario.dart';
+import '../robots/composer_robot.dart';
+import '../robots/search_robot.dart';
+import '../robots/thread_robot.dart';
+import 'login_with_basic_auth_scenario.dart';
+
+class SearchResultHighlightsScenario extends BaseScenario {
+  SearchResultHighlightsScenario(
+    super.$, {
+    required this.loginWithBasicAuthScenario,
+    required this.keyword,
+    required this.longContentWithSearchKeywordAtTheStart,
+    required this.longContentWithSearchKeywordAtTheMiddle,
+    required this.longContentWithSearchKeywordAtTheEnd,
+  });
+
+  final LoginWithBasicAuthScenario loginWithBasicAuthScenario;
+  final String keyword;
+  final String longContentWithSearchKeywordAtTheStart;
+  final String longContentWithSearchKeywordAtTheMiddle;
+  final String longContentWithSearchKeywordAtTheEnd;
+
+  @override
+  Future<void> execute() async {
+    // Robots
+    final threadRobot = ThreadRobot($);
+    final composerRobot = ComposerRobot($);
+    final searchRobot = SearchRobot($);
+
+    // Login
+    await loginWithBasicAuthScenario.execute();
+
+    // Send some emails
+    await _sendEmail(
+      threadRobot: threadRobot,
+      composerRobot: composerRobot,
+      subject: keyword,
+      content: longContentWithSearchKeywordAtTheStart,
+      grandContactPermission: true,
+    );
+    await _sendEmail(
+      threadRobot: threadRobot,
+      composerRobot: composerRobot,
+      subject: keyword,
+      content: longContentWithSearchKeywordAtTheMiddle,
+    );
+    await _sendEmail(
+      threadRobot: threadRobot,
+      composerRobot: composerRobot,
+      subject: keyword,
+      content: longContentWithSearchKeywordAtTheEnd,
+    );
```

Comment:
We can use `for` loop for it with `List<String> longContent`

Reply:
Done